powerauth-push-client/src/main/java/io/getlime/push/client/PushServerClient.java

```diff
@@ -121,15 +121,17 @@ public boolean createDevice(String appId, String token, MobilePlatform platform)
      * @param appId PowerAuth application app ID.
      * @param token Token received from the push service provider (APNs, FCM).
      * @param platform Mobile platform (APNs, FCM, HMS).
+     * @param environment APNs environment (optional).
      * @param activationId PowerAuth activation ID.
      * @return True if device registration was successful, false otherwise.
      * @throws PushServerClientException In case of network, response / JSON processing, or other IO error.
      */
-    public boolean createDevice(String appId, String token, MobilePlatform platform, String activationId) throws PushServerClientException {
+    public boolean createDevice(String appId, String token, MobilePlatform platform, ApnsEnvironment environment, String activationId) throws PushServerClientException {
```

Comment:
Strictly speaking, this is not binary compatible; on the other hand, it is easy for the clients to fix. Anyway, I suggest deprecating these methods and introducing a request parameter. Could be handled by a different PR.

Reply:
Ok, I can make the change in a compatible way and use deprecation and write a migration note.